File: assignment/q2/q2.java
// package assignment.q2;

import java.util.*;

public class q2 {
    public static void main(String args[]) {
        q2 ob = new q2();
        Scanner sc = new Scanner(System.in);
        double temp;
        System.out.println("Enter Room Temperature");

        try {
            temp = sc.nextDouble();

            if (temp < 18.0) {
                throw new Exception("Room temperature is too low");
            } else if (temp > 28.0) {
                throw new Exception("Room temperature is too high");
            } else {
                System.out.println("Valid Room Temperature: " + temp);
            }
        } catch (InputMismatchException e) {
            System.out.println("Invalid input! Please enter a numeric value.");
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }

        sc.close();
    }
}

File: assignment/q2/q2.cpp
#include <iostream>
#include <stdexcept>

using namespace std;

int main()
{
    double temp;
    cout << "Enter Room Temperature: ";

    try
    {
        cin >> temp;

        if (!cin)
        {
            throw runtime_error("Invalid input. Please enter a numeric value.");
        }

        if (temp < 18.0)
        {
            throw runtime_error("Room temperature is too low");
        }
        else if (temp > 28.0)
        {
            throw runtime_error("Room temperature is too high");
        }
        else
        {
            cout << "Valid Room Temperature: " << temp << endl;
        }
    }
    catch (const runtime_error &e)
    {
        cout << e.what() << endl;
    }

    return 0;
}
